src/wazuh_db/wdb_agent.c

```diff
@@ -379,6 +380,61 @@ int wdb_update_agent_keepalive(int id, const char *sync_status, int *sock) {
     return result;
 }
 
+int wdb_update_agent_connection_status(int id, const char *connection_status, int *sock) {
+    int result = 0;
+    cJSON *data_in = NULL;
+    char *data_in_str = NULL;
+    char *wdbquery = NULL;
+    char *wdboutput = NULL;
+    char *payload = NULL;
+    int aux_sock = -1;
+
+    data_in = cJSON_CreateObject();
+
+    if (!data_in) {
+        mdebug1("Error creating data JSON for Wazuh DB.");
+        return OS_INVALID;
+    }
+
+    cJSON_AddNumberToObject(data_in, "id", id);
+    cJSON_AddStringToObject(data_in, "connection_status", connection_status);
+    data_in_str = cJSON_PrintUnformatted(data_in);
+
+    os_calloc(WDBQUERY_SIZE+1, sizeof(char), wdbquery);
+    os_calloc(WDBOUTPUT_SIZE, sizeof(char), wdboutput);
+    snprintf(wdbquery, WDBQUERY_SIZE, global_db_commands[WDB_UPDATE_AGENT_KEEPALIVE], data_in_str);
```

Comment:
Here we should use the **WDB_UPDATE_AGENT_CONNECTION_STATUS** command instead.

Reply:
FIXED